news/2022-08-25-changes-to-osu-taiko-sr-and-pp.md

```diff
@@ -0,0 +1,149 @@
+
+- Instead of assuming a full-alternate play style, we assume that ****each note of the same colour is always alternated.** For example, for the pattern kdkdkd, we assume the kat's will be hit by **k<sup>1</sup>** d<sup>1</sup> d<sup>2</sup> **k<sup>2</sup>** d<sup>1</sup> d<sup>2</sup> **k<sup>1</sup>**, and the don's will be hit by k<sup>1</sup> **d<sup>1</sup> d<sup>2</sup>** k<sup>2</sup> **d<sup>1</sup> d<sup>2</sup>** k<sup>2</sup>.
+
+- The reason for this is to assume the best case scenario for repeated keystrokes regardless of play style. In other words, it aims to approximate the theoretical minimum stamina required to play a given pattern with any playstyle, with any theoretically rollable pattern being rolled. We can’t (feasibly), nor is it desired to, detect playstyles, so we assume the case that requires the least physical repetition, which is to switch keys each time a colour is to be hit, and each play style will have their respective weaknesses.
+
+This also helps with the cases where specific playstyles or patterns were nerfed in the previous rework, as we don't detect any patterns, just the pure stamina requirement for each map.
+
+## Readdressing Colour
+
+The old colour system detected the frequency of colour changes and had penalties for specific repeated patterns, this meant that only notes with a different colour from the previous could get more than a non-zero strain, which contributed to issues of complex patterns containing large groups of single coloured notes not being considered properly in the strain system.
+
+Similar to stamina, a repetition penalty was applied on durations of single-coloured patterns. While only the 5 most recent durations were considered in this penalty, if any two or more of those patterns were repeated, a harsh penalty was applied.
+
+Once again, going against the new policies of this rework, changes were made.
+
+#### The new Colour system
+
+For the new system we define colour difficulty as the amount of information contained within a given pattern (or sequence if you will). For example, we can take a look with the following colour patterns and the very rough amount of information required to describe them:
+
+![](/wiki/shared/news/2022-08-25-changes-to-osu-taiko-star-rating-and-performance-points/colour-graphic-1.jpg)
+
+As we can see, a more complex pattern (by definition) requires more information to describe. To compute this, we run a compression algorithm to try to encode a given colour sequence with the minimum amount of information required. One thing to note is traditional compression algorithms typically build a dictionary from repetitive patterns, the new algorithm however, has to take where the repetitions take place into account, as repetitions that occur far from each other are typically harder to play than repetitions that’s close to each other. 
```

Comment:
smart quotes bad 
```suggestion
As we can see, a more complex pattern (by definition) requires more information to describe. To compute this, we run a compression algorithm to try to encode a given colour sequence with the minimum amount of information required. One thing to note is traditional compression algorithms typically build a dictionary from repetitive patterns, the new algorithm however, has to take where the repetitions take place into account, as repetitions that occur far from each other are typically harder to play than repetitions that's close to each other. 
```

Reply:
marktext is failing me, sorted in 1fa2745